std/range/package.d

```diff
@@ -9594,6 +9594,324 @@ auto refRange(R)(R* range)
     foo(r);
 }
 
+/**
+Bitwise adapter over Integral type Ranges. Consumes the range elements bit by bit.
+*/
+struct Bitwise(R)
+    if (isInputRange!R && isIntegral!(ElementType!R))
+{
+    alias ER = ElementType!R;
+    alias UER = Unsigned!ER;
+
+    private R parent;
+    private UER mask;
+    private int maskPos;
+    private ER accumulator;
+
+    private enum UER bitsNum = ER.sizeof * 8;
+
+    this()(auto ref R range)
+    {
+        parent = range;
+
+        static if (hasLength!R)
+        {
+            len = parent.length * bitsNum;
+        }
+
+        initPrivates();
+
+        static if (isBidirectionalRange!R)
+        {
+            initBackPrivates();
+        }
+    }
+
+    bool empty() const
+    {
+        static if (hasLength!R)
+        {
+            return len == 0;
+        }
+        else
+        {
+            /*
+               If we consumed the last element of the range, but not all the
+               bits in the last element
+             */
+            return parent.empty && (maskPos == 0);
+        }
+    }
+
+    bool front()
+    {
+        return cast(bool)(accumulator & mask);
+    }
+
+    void popFront()
+    {
+        if (mask == 0)
+        {
+            initPrivates();
+        }
+        else
+        {
+            mask >>>= 1;
+            --maskPos;
+        }
+
+        --len;
+    }
+
+    private void initPrivates()
+    {
+        // Helper function to avoid code duplication
+        mask = cast(UER)(1) << (bitsNum - 1);
+        maskPos = bitsNum;
+        accumulator = parent.front;
+        parent.popFront;
+    }
+
+    static if (hasLength!R)
+    {
+        private size_t len;
+
+        size_t length() const
+        {
+            return len;
+        }
+
+        alias opDollar = length;
+    }
+
+    static if (isForwardRange!R)
+    {
+        typeof(this) save()
+        {
+            return this;
+        }
+    }
+
+    static if (isBidirectionalRange!R)
+    {
+        private ER backMask;
+        private int backMaskPos;
+        private ER backAccumulator;
+
+        private void initBackPrivates()
+        {
+            // Helper function to avoid code duplication
+            if (!parent.empty)
+            {
+                backMask = 1;
+                backMaskPos = 1;
+                backAccumulator = parent.back;
+                parent.popBack;
+            }
+        }
+
+        bool back()
+        {
+            return cast(bool)(backAccumulator & backMask);
```

Comment:
to avoid using `cast` and triggering all greps for unsafe code, use `return (backAccumulator & backMask) != 0;`

Also, here you can use `const` on the method because you don't depend on any templated stuff.